x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/plan/logical/EsRelation.java

```diff
@@ -34,7 +34,7 @@ public EsRelation(Source source, EsIndex index, List<Attribute> attributes) {
         this(source, index, attributes, false);
     }
 
-    private EsRelation(Source source, EsIndex index, List<Attribute> attributes, boolean frozen) {
+    public EsRelation(Source source, EsIndex index, List<Attribute> attributes, boolean frozen) {
```

Comment:
Serialization should recreate an equivalent instance, so needs access to this constructor.